Source/CesiumRuntime/Private/CesiumGltfComponent.cpp

```diff
@@ -78,33 +78,9 @@ using TMeshVector4 = FVector4f;
 static uint32_t nextMaterialId = 0;
 
 namespace {
-void destroyHalfLoadedTexture(
-    TUniquePtr<CesiumTextureUtility::LoadedTextureResult>& pHalfLoadedTexture) {
-  if (pHalfLoadedTexture) {
-    CesiumTextureUtility::destroyHalfLoadedTexture(*pHalfLoadedTexture.Get());
-  }
-}
 class HalfConstructedReal : public UCesiumGltfComponent::HalfConstructed {
```

Comment:
I'm confused on if we _need_ this `HalfConstructedReal` class anymore? Could it be merged with its base class `UCesiumGltfComponent::HalfConstructed`?

Reply:
It's a compiler firewall. `CreateOffGameThread` needs to be able to return a thing so that it can be passed to `CreateOnGameThread` and so that it can be freed later. But the actual contents of that thing are an implementation detail. If we were to put the actual contents of `HalfConstructedReal` into `HalfConstructed`, then any source file including CesiumGltfComponent.h would (by necessity) have a full declaration of all the contents of that class, which would require including a whole bunch more headers. So this approach makes compilation a bit faster; we have to recompile fewer source files and less often.

Reply:
That makes sense, thanks! :pray: